### ISSUE TITLE
bug: ssue Title: Parser incorrectly fails on &ident[..] syntax when identifier is named raw

### ISSUE THREAD

**Issue by 99memory** [tree-sitter/tree-sitter-rust#277]
### Did you check existing issues?

- [x] I have read all the [tree-sitter docs](https://tree-sitter.github.io/tree-sitter/using-parsers) if it relates to using the parser
- [x] I have searched the existing issues of tree-sitter-rust

### Describe the bug

The parser incorrectly flags a valid piece of Rust code as a syntax error. The error seems to be specifically triggered when a variable named raw is used in a slice-and-compare context like &raw[0..4].

Renaming the variable to anything else (e.g., rww) makes the parsing error disappear. This suggests a bug in how the parser handles the identifier raw in this specific grammatical structure, likely treating it as a keyword by mistake.

Steps to Reproduce
Create a Rust file (test.rs) with the following minimal code containing two functions.

### Steps To Reproduce/Bad Parse Tree

`// Minimal code to demonstrate the bug

const TAG: [u8; 4] = [0, 1, 2, 3];

// 1. This function fails to parse
fn check_fails(raw: &Vec<u8>) {
    if &raw[0..4] != TAG {
        // ...
    }
}

// 2. This function parses correctly
fn check_works(rww: &Vec<u8>) {
    if &rww[0..4] != TAG {
        // ...
    }
}`

### Expected Behavior/Parse Tree

- File {
    shebang: undefined
    attrs: []
- items: [
    + ItemConst {attrs, vis, const_token, ident, colon_token, ... +4}
    + ItemFn {attrs, vis, sig, block}
    - ItemFn {
        attrs: []
        vis: Visibility::Inherited {}
        + sig: Signature {constness, asyncness, unsafety, abi, fn_token, ... +6}
        - block: Block {
            + stmts: [1 element]
            brace_token: Brace {
            }
          }
      }
  ]
}

### Repro

```Rust
// Example code that causes the issue
fn foo() {
  // Code that fails to parse, or causes an error
}
```

**Comment by wetneb** [comment]
Curious… `raw` isn't excluded anywhere from the set of valid identifiers, so I'm not clear on why it's happening, but since primitive types are explicitly added as valid identifiers, I think we can do the same for `raw` (and possibly a bunch of others?).
I have made [a fix for it](https://codeberg.org/grammar-orchard/tree-sitter-rust-orchard/commit/35cf85800fee95163c33a49099cc2d984e09a163), not sure if it's the right approach though.

Maintainers of this repository aren't active currently, but if it's an option for you, you can benefit from the fix by switching to [the fork](https://codeberg.org/grammar-orchard/tree-sitter-rust-orchard) (if you install the grammar from other places than crates.io, let me know and I'd be happy to look into publishing the fork to the package repository).
